docs/open_source/scan/scan_vision/index.md

```diff
@@ -17,68 +17,333 @@ pip install giskard giskard-vision
 
 ## Step 1: Wrap your dataset
 
+To scan your model, start by **wrapping your dataset**, as shown below
+
+:::::::{tab-set}
+::::::{tab-item} Image classification
+
+To scan your model, start by **wrapping your dataset** with `DataLoaderBase`. Your class should override:
+
+- `idx_sampler` (property) that returns the array of indices to iterate over
+- `get_image` that loads the image corresponding to the right index as a `np.ndarray`
+- `get_labels` that returns the label of the image with corresponding index
+- (Optional) `get_meta` that returns a `MetaData` object containing the metadata of the image with the corresponding index. This object should contain the metadata in the form of a `dict`, for instance `{'meta1': 'value1'}`, the list of categorical metadata, and the type of Issue related to the metadata in the form of a `dict`, for instance `{'meta1': PerformanceIssueMeta}`. Issue types can be found in `giskard_vision.core.issues`.
+
+> ### ⚠️ Warning
+>
+> It's highly recommended that you wrap your data **before preprocessing** so that you can easily interpret
+> the scan results.
+>
+> For now, the image classification scan only supports one label per image
+>
+
+```python
+from giskard_vision.core.dataloaders.base import DataIteratorBase
+from giskard_vision.core.dataloaders.meta import MetaData
+from giskard_vision.core.issues import EthicalIssueMeta, PerformanceIssueMeta
+
+
+class DataLoaderClassification(DataIteratorBase):
+
+    @property
+    def idx_sampler(self) -> np.ndarray:
+        """Accesses the array of element indices to iterate over
+
+        Returns:
+            np.ndarray: The array of indices
+        """
+        ...
+
+    @classmethod
+    def get_image(self, idx: int) -> np.ndarray:
+        """
+        Returns a single image from an index as numpy array
+
+        Args:
+            idx (int): Index of the image
+
+        Returns:
+            np.ndarray: The image as numpy array (h, w, c)
+        """
+        return cv2.imread(images_paths[idx])
+
+    @classmethod
+    def get_labels(self, idx: int) -> Optional[np.ndarray]:
+        """
+        Gets the label (for a single image) for a specific index.
+
+        Args:
+            idx (int): Index of the image.
+
+        Returns:
+            Optional[np.ndarray]: Label for the given index.
+        """
+        return np.array(..., dtype=float)
```

Comment:
we support only single-label classification, the expected outcome of this cannot be anything but a one-value string